src/HttpHeaders.php

```diff
@@ -0,0 +1,44 @@
+<?php
+
+namespace P4\MasterTheme;
+
+/**
+ * Class HttpHeaders
+ */
+class HttpHeaders {
+	/**
+	 * Headers constructor.
+	 */
+	public function __construct() {
+		add_action( 'wp_headers', [ $this, 'send_content_security_policy_header' ], 10, 1 );
+	}
+
+
+	/**
+	 * Send Content Security Policy (CSP) HTTP headers.
+	 */
+	public function send_content_security_policy_header() {
+		$allowed_frame_ancestors = [ '\'self\'' ];
+
+		// Filter hook to allow modification of trusted frame ancestors.
+		$modified_allowed_frame_ancestors = apply_filters( 'planet4_csp_allowed_frame_ancestors', $allowed_frame_ancestors );
+
+		if ( ! is_array( $modified_allowed_frame_ancestors ) ) {
+			$modified_allowed_frame_ancestors = $allowed_frame_ancestors;
```

Comment:
Since this variable is used further in the code, it's more convenient to just call it `$allowed_frame_ancestors`, and also more accurate, since it might not be modified. We can give a lengthy name to the base config instead.

Then catching the case of the filter returning an invalid value becomes more readable
```php
if ( ! is_array( $allowed_frame_ancestors ) ) {
			$allowed_frame_ancestors = $default_allowed_frame_ancestors;